ccdproc/ccddata.py

```diff
@@ -279,43 +279,66 @@ def _ccddata_arithmetic(self, other, operation, scale_uncertainty=False):
                          meta=new_meta, wcs=new_wcs)
         return result
 
-    def multiply(self, other, compare_wcs=False):
+    def multiply(self, other, compare_wcs='first_found'):
         if isinstance(other, CCDData):
-            nother = other.copy()
-            if not compare_wcs:
+            if compare_wcs is None or compare_wcs == 'first_found':
+                nother = other.copy()
                 nother.wcs = self.wcs
-            return super(CCDData, self).multiply(nother)
+                return super(CCDData, self).multiply(nother)
+            else:
+                try:
+                    return super(CCDData, self).multiply(
+                        other, compare_wcs=compare_wcs)
+                except:
+                    raise ValueError('WCS can not be compared')
 
         return self._ccddata_arithmetic(other, np.multiply,
                                         scale_uncertainty=True)
 
-    def divide(self, other, compare_wcs=False):
+    def divide(self, other, compare_wcs='first_found'):
         if isinstance(other, CCDData):
-            nother = other.copy()
-            nother.wcs = self.wcs
-            if not compare_wcs:
+            if compare_wcs is None or compare_wcs == 'first_found':
+                nother = other.copy()
                 nother.wcs = self.wcs
-            return super(CCDData, self).divide(nother)
+                return super(CCDData, self).divide(nother)
+            else:
+                try:
+                    return super(CCDData, self).divide(other,
```

Comment:
For astropy 1.1.2 and 1.0.9 the upstream-fix (https://github.com/astropy/astropy/pull/4503) allows comparing WCS objects but will raise a ``TypeError`` in case one tries to give it a ``compare_wcs`` (which only comes for astropy 1.2). That's the reason why I chose a different approach in #292.